test/specs/validate-spec/valid/3.0/component-schema-with-hyphens.yaml

```diff
@@ -0,0 +1,25 @@
+openapi: 3.0.3
+info:
+  version: "1.0"
+  title: Invalid API
+paths:
+  /anything:
+    get:
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  "$ref": "#/components/schemas/User-Information"
+components:
+  schemas:
+    User-Information:
```

Comment:
Hyphens are allowed per the spec (https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#components-object), this is just to ensure that the regex we're asserting against is allowing them.